newdle/api.py

```diff
@@ -323,36 +324,68 @@ def create_newdle(title, duration, timezone, timeslots, participants, private, n
             ),
         },
     )
+    db.session.commit()
     return NewdleSchema().jsonify(newdle)
 
 
-@api.route('/newdle/<code>')
-@allow_anonymous
-def get_newdle(code):
-    newdle = Newdle.query.filter_by(code=code).first_or_404(
-        'Specified newdle does not exist'
-    )
-    if newdle.deleted:
-        return DeletedNewdleSchema().jsonify(newdle)
-    return RestrictedNewdleSchema().jsonify(newdle)
-
-
 @api.route('/newdle/<code>', methods=('PATCH',))
-@use_args(UpdateNewdleSchema(), locations=('json',))
+@use_args(NewNewdleSchema(partial=True), locations=('json',))
 def update_newdle(args, code):
     newdle = Newdle.query.filter_by(code=code).first_or_404(
         'Specified newdle does not exist'
     )
-    if newdle.creator_uid != g.user['uid']:
+    if g.user is None or newdle.creator_uid != g.user['uid']:
```

Comment:
`g.user is None` should never happen here since the code in `before_request` aborts early for endpoints without `allow_anonymous`

Reply:
Makes sense, had the impression we had it before.

Reply:
that check still isn't needed ;)